### ISSUE TITLE
Provide for a simpler way to open the new App Menu with a keyboard shortcut

### ISSUE THREAD

**Issue by UndeadDevel** [QubesOS/qubes-issues#9005]
### The problem you're addressing (if any)
Right now the way to open the new App Menu with a terminal or via a keyboard shortcut is by making the following call:
```
gdbus call --session --dest org.qubesos.appmenu --object-path "/org/qubesos/appmenu" --method org.freedesktop.Application.Activate "{}"
```
To avoid various issues, even more "hacks" are required, see [my current script](https://gist.github.com/UndeadDevel/8501e82a534e5daa2ac60e36945167b9).

### The solution you'd like
This should be easier and there should be a standard keyboard shortcut already set up.

In other words, since the new App Menu seems to aim to replace not just the old App Menu, but also the Xfce Application Finder, which can be called with `xfce4-appfinder` and has a standard shortcut set up on installation, there should be some wrapper "program" (script) with an already set up keyboard shortcut for it as well, perhaps simply replacing the Xfce one.

### The value to a user, and who that user might be
Better integrated App Menu, especially for users who use the keyboard more or primarily.

**Comment by marmarta** [comment]
This will be doable once dom0 moves to newer xfce, because we need this fix: https://gitlab.xfce.org/xfce/libxfce4ui/-/issues/1#note_12979

**Comment by marmarta** [comment]
To be exact: we need xfce version at least 4.18.5

**Comment by ben-grande** [comment]
> To be exact: we need xfce version at least 4.18.5

Dom0 f41 R4.3 has Xfce 4.18.6

**Comment by marmarek** [comment]
I tried it recently on R4.3. It's "almost" good: now assigning super key to the menu works - pressing just super launches the menu, but pressing for example super+l locks the screen (per my custom configuration). But, key combos of xfwm with super stop working (for example keys for switching workspaces, maximizing/minimizing windows etc - if configured to be super+something).

So, we can't have just "Super" to launch the menu... at least not yet. But we can have some other combo - for example replace the Alt+F1 that currently launches the desktop menu (same as right click on the desktop). Or choose some new, yet simple combo.

**Comment by parulin** [comment]
I use `Super`+`Q`, because of the icon of the app menu.

**Comment by ben-grande** [comment]
> I use `Super`+`Q`, because of the icon of the app menu.

Checked on currently supported WMs, Xfce, KDE and i3.

- i3: free
- Xfce: free
- KDE: In use by some activity monitor on the side

I still think `Super+q` is the best, so maybe we should override the KDE variant.

**Comment by marmarek** [comment]
One problem with Super+Q is that "Q" is usually for "quit", and many applications use Ctrl+Q for closing - which is rather close and may be easy to mistype. Similarly, Super+Shift+Q in i3 kills the current window, which is also undesirable if you want to open the menu...
@marmarta what do you think?

**Comment by ben-grande** [comment]
Maybe we don't need to worry of i3 because you added `$mod+shift+d` to open the app menu: https://github.com/QubesOS/qubes-desktop-linux-i3-settings-qubes/commit/f3fe6adcf343d4b1495fa7e10507d793164d5cc7.

So let me retract i3 from the table and just focus on KDE and Xfce.



> I tried it recently on R4.3. It's "almost" good: now assigning super key to the menu works - pressing just super launches the menu, but pressing for example super+l locks the screen (per my custom configuration). But, key combos of xfwm with super stop working (for example keys for switching workspaces, maximizing/minimizing windows etc - if configured to be super+something).
> 
> So, we can't have just "Super" to launch the menu... at least not yet. But we can have some other combo - for example replace the Alt+F1 that currently launches the desktop menu (same as right click on the desktop). Or choose some new, yet simple combo.

Can you please retest this?

I just tested it and it works:

- `xfce4-session-logout`: `Super+L` -> This is "Supler plus 'elle'"
- `qubes-app-menu`: `Super L` -> this is "Super of the Left"

I did via GUI, but via CLI, it might be:
```
xfconf-query -c xfce4-keyboard-shortcuts -l
xfconf-query -c xfce4-keyboard-shortcuts -p "/commands/custom/<Super>l" -n -s "xfce4-session-logout" -t string
xfconf-query -c xfce4-keyboard-shortcuts -p "/commands/custom/Super_L" -n -s "qubes-app-menu" -t string
```

The app menu appears to be following the mouse, but that is another issue. Both combinations work, the difference is, for "Super_L", you press and release, to run the shortcut, for "<Super>l", it awaits for you to finish the combination or to release the "<Super>" to decide what to do.

**Comment by marmarek** [comment]
> Can you please retest this?

Read my comment again ;) The thing you described does work, but try using Super+something in xfwm settings now (System settings -> Window Manager -> Keyboard tab)

**Comment by ben-grande** [comment]
What about `Super+Alt L`? I tested and it doesn't break xfwm4 shortcuts that have Super.

**Comment by marmarta** [comment]
I mean, but this is supremely undiscoverable (you have to poke around in documentation); with Q, at least you have a Q on the button. I'm not opposed to other solutions, but there needs to be a way to make them discoverable that comes with them.

**Comment by andrewdavidwong** [comment]
> I mean, but this is supremely undiscoverable (you have to poke around in documentation); with Q, at least you have a Q on the button. I'm not opposed to other solutions, but there needs to be a way to make them discoverable that comes with them.

Sounds like an explanatory tooltip (#2211) might help here: hover over the button, and it shows the keyboard shortcut.

**Comment by marmarta** [comment]
Tooltip can be done (in XFCE). I would still like something that is a bit more understandable, although I see y'alls arguments on Super+Q. Super+M, for Menu?

**Comment by ben-grande** [comment]
> Super+M, for Menu?

This seems safer, yes.

**Comment by parulin** [comment]
Super+M is fine but it is a bit hard to use with only one hand. I don't know how to check if Super+A(pp menu) is already mapped.
